llmfoundry/models/hf/hf_fsdp.py

```diff
@@ -86,13 +88,13 @@ def hf_get_hidden_layers(model: PreTrainedModel) -> Any:
         - transformer.blocks: (MPTForCausalLM)
     """
     hidden_layers_attrs = (
-        'transformer.h',  # BLOOM, GPT2, GPTJ
-        'model.decoder.layers',  # OPT
-        'gpt_neox.layers',  # GPTNeoX
+        'h',  # BLOOM, GPT2, GPTJ
+        'decoder.layers',  # OPT
+        'layers',  # GPTNeoX
         'block',  # T5, BART, Pegasus (from encoder)
         'layers',  # ProphetNet, Marian (from encoder)
-        'model.layers',  # LLaMa
-        'transformer.blocks',  # MPT
+        'layers',  # LLaMa
+        'blocks',  # MPT
```

Comment:
why dont' we need the prefixes anymore?

Reply:
I refactored below for this func to take in `causal_base_model` instead of `model`, cause it was cleaner that way.

Reply:
updated the docstring

Reply:
any reason why `layers` is repeated?

Reply:
cause i didn't notice it was the same after removing the prefixes :)

Reply:
fixed